### PR TITLE
GEODE-7425 delete index

### PR DESCRIPTION
**Pull request by dschneider-pivotal**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-web-management/src/main/java/org/apache/geode/management/internal/rest/controllers/RegionManagementController.java [hunk 1]
```diff
@@ -196,4 +196,20 @@ public ClusterManagementResult deleteRegion(
     return new ResponseEntity<>(result,
         HttpStatus.CREATED);
   }
+
+  @ApiOperation(value = "delete region index")
+  @PreAuthorize("@securityService.authorize('CLUSTER', 'MANAGE', 'QUERY')")
+  @DeleteMapping(REGION_CONFIG_ENDPOINT + "/{regionName}" + INDEXES + "/{indexName}")
```
Comment (jinmeiliao) [inline geode-web-management/src/main/java/org/apache/geode/management/internal/rest/controllers/RegionManagementController.java]: Would the index name have "." in it? Should we add that :. thing like other end points?

File: geode-web-management/src/main/java/org/apache/geode/management/internal/rest/controllers/RegionManagementController.java [hunk 2]
```diff
@@ -196,4 +196,20 @@ public ClusterManagementResult deleteRegion(
     return new ResponseEntity<>(result,
         HttpStatus.CREATED);
   }
+
+  @ApiOperation(value = "delete region index")
+  @PreAuthorize("@securityService.authorize('CLUSTER', 'MANAGE', 'QUERY')")
```
Comment (jinmeiliao) [inline geode-web-management/src/main/java/org/apache/geode/management/internal/rest/controllers/RegionManagementController.java]: would be nice to add an entry in the "ClusterManagementSecurityRestIntegrationTest" to test this.

File: geode-web-management/src/main/java/org/apache/geode/management/internal/rest/controllers/RegionManagementController.java (hunk 1 above)
Comment (dschneider-pivotal) [reply geode-web-management/src/main/java/org/apache/geode/management/internal/rest/controllers/RegionManagementController.java]: Good catch. We should do that for all endpoints that end with a parameter

Comment (jmelchio) [reply geode-web-management/src/main/java/org/apache/geode/management/internal/rest/controllers/RegionManagementController.java]: Updated the code to cover this scenario.

File: geode-web-management/src/main/java/org/apache/geode/management/internal/rest/controllers/RegionManagementController.java (hunk 2 above)
Comment (jmelchio) [reply geode-web-management/src/main/java/org/apache/geode/management/internal/rest/controllers/RegionManagementController.java]: Added context for testing permissions in `ClusterManagementSecurityRestIntegrationTest`.

**Comment by jmelchio** [comment]
@demery-pivotal please review

File: geode-web-management/src/main/java/org/apache/geode/management/internal/rest/controllers/RegionManagementController.java [hunk 3]
```diff
@@ -196,4 +196,20 @@ public ClusterManagementResult deleteRegion(
...
+  @DeleteMapping(REGION_CONFIG_ENDPOINT + "/{regionName}" + INDEXES + "/{indexName:.+}")
```
Comment (dschneider-pivotal) [inline geode-web-management/src/main/java/org/apache/geode/management/internal/rest/controllers/RegionManagementController.java]: Did you consider adding a test to verify ":.+" is working to RegionManagementControllerSpringTest? Currently that is all that test does so you should be able to copy one of its current methods to cover this endpoint

File: geode-web-management/src/main/java/org/apache/geode/management/internal/rest/controllers/RegionManagementController.java [hunk 4]
```diff
@@ -196,4 +196,20 @@ public ClusterManagementResult deleteRegion(
...
+  public ClusterManagementResult deleteRegion(
```
Comment (dschneider-pivotal) [inline geode-web-management/src/main/java/org/apache/geode/management/internal/rest/controllers/RegionManagementController.java]: Shouldn't this method be named deleteIndex?

Comment (jmelchio) [reply geode-web-management/src/main/java/org/apache/geode/management/internal/rest/controllers/RegionManagementController.java]: Yes. I'll make that change.

File: geode-web-management/src/main/java/org/apache/geode/management/internal/rest/controllers/RegionManagementController.java (hunk 3 above)
Comment (jmelchio) [reply geode-web-management/src/main/java/org/apache/geode/management/internal/rest/controllers/RegionManagementController.java]: We did make a change to the DUnit test to do the same and that will exercise the endpoint. I can add one to the `RegionManagementControllerSpringTest` as well.

**Comment by jmelchio** [comment]
@demery-pivotal good feedback. I'll update the code to address the issues.

File: geode-assembly/src/distributedTest/java/org/apache/geode/management/internal/rest/ListIndexManagementDUnitTest.java [hunk 5]
```diff
@@ -52,12 +53,14 @@
...
   private static ClusterManagementService cms;
+  private static MemberVM locator;
 
   @BeforeClass
   public static void beforeClass() {
-    MemberVM locator = lsRule.startLocatorVM(0, MemberStarterRule::withHttpService);
+    locator = lsRule.startLocatorVM(0, MemberStarterRule::withHttpService);
     MemberVM server1 = lsRule.startServerVM(1, locator.getPort());
     MemberVM server2 = lsRule.startServerVM(2, locator.getPort());
+    MemberVM server3 = lsRule.startServerVM(3, "group1", locator.getPort());
```
[the remark this answers is not in the document]
Comment (jmelchio) [reply geode-assembly/src/distributedTest/java/org/apache/geode/management/internal/rest/ListIndexManagementDUnitTest.java]: > Please name the members to indicate their purpose in the test. For example, t looks as if the key idea for `server3` is that it is in a group. Name both the variable and the server to indicate that intent. That way, when someone reads the test, they have a better chance of understanding why this variable or string is used as an input to something, or should appear in a result.

I'm not sure if we can name the server. I believe the server name is simply the word `server` with the number that is given in the `startServer` method of the rule. I can certainly change the name of the variable.

File: geode-assembly/src/distributedTest/java/org/apache/geode/management/internal/rest/ListIndexManagementDUnitTest.java [hunk 6]
```diff
@@ -226,4 +229,85 @@ public void listNonExistingIndexesWithIdFilter() {
       softly.assertThat(list.isSuccessful()).isTrue();
     });
   }
+
+  @Test
+  public void createAndDeleteIndex_success_in_specific_group() {
+    Region config = new Region();
```
[the remark this answers is not in the document]
Comment (jmelchio) [reply geode-assembly/src/distributedTest/java/org/apache/geode/management/internal/rest/ListIndexManagementDUnitTest.java]: > Consider: If a variable represents a filter, name it to identify it as a filter. Like `regionFilter` and `indexFilter`. If a variable represents a config, name it to identify it as a config, like `regionConfig` and `indexConfig`.

A bit of a challenge here is that in some cases we use the variable to create something first, and as a filter to fetch to confirm the creation later.

File: geode-assembly/src/distributedTest/java/org/apache/geode/management/internal/rest/ListIndexManagementDUnitTest.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (demery-pivotal) [reply geode-assembly/src/distributedTest/java/org/apache/geode/management/internal/rest/ListIndexManagementDUnitTest.java]: Maybe in that case we can find a name that encompasses both intentions. Maybe something like `configSpecification`.

**Pull request merged**